functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,196 @@ public static void composeObject_test6() throws Exception {
     }
   }
 
+  /**
+   * Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)
+   * disableObjectLegalHold(String bucketName, String objectName, String versionId)
+   */
+  public static void enableDisableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)");
+      System.out.println(
+          "Test: disableObjectLegalHold(String bucketName, String objectName, String versionId)");
```

Comment:
we would need to have two separate tests here.
1. `enableObjectLegalHold()` - enable legal hold on an object; on success, overwrite/delete object should fail
2. `disableObjectLegalHold()` - disable legal hold on an object; on success, overwrite/delete object should succeed